hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieInstant.java

```diff
@@ -136,7 +136,10 @@ public HoodieInstant(StoragePathInfo pathInfo) {
           state = State.COMPLETED;
         }
       }
-      completionTime = timestamps.length > 1 ? timestamps[1] : null;
+      completionTime = timestamps.length > 1
+          ? timestamps[1]
+          // for backward compatibility with 0.x release.
+          : state == State.COMPLETED ? pathInfo.getModificationTime() + "" : null;
```

Comment:
Could we print a warning message on such an instant to make sure the user is aware of this since the modification time is not a robust as the completion time?

Reply:
The legacy release already uses the modification time in conflcits resolution.